internal/minipipeline/analysis.go

```diff
@@ -377,6 +382,16 @@ func (wa *WebAnalysis) dnsComputeFailureMetrics(c *WebObservationsContainer) {
 		// TODO(bassosimone): if we set an IPv6 address as the resolver address, we
 		// end up with false positive errors when there's no IPv6 support
 
+		// lookups once we started following redirects does not have corresponding
+		// control information, so failures end up being unexplained
+		if obs.TagDepth.IsNone() || obs.TagDepth.Unwrap() != 0 {
+			if obs.DNSLookupFailure.Unwrap() != "" {
+				wa.DNSLookupUnexplainedFailure.Add(obs.DNSTransactionID.Unwrap())
+				continue
+			}
+			continue
+		}
```

Comment:
Moving this block of code down here should be fine, given that we're just moving checks that cause us to skip processing above and we end up skipping the entry with `continue` in any case in the `if` block.